Query PostgreSQL money type cause ERROR: Bad value for type double : -92,233,720,368,547,758.08

## Bug Report

### Which version of ShardingSphere did you use?

master - a4c01712aba7f17f3f4fe8264aaf4c19cb572432

This issue may affect all version of ShardingSphere-Proxy.

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

ShardingSphere-Proxy PostgreSQL

### Expected behavior

Query worked.

Expected output:
```
-$92,233,720,368,547,758.08
```

### Actual behavior

Client:
```
Exception in thread "main" org.postgresql.util.PSQLException: ERROR: Bad value for type double : -92,233,720,368,547,758.08
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2675)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2365)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:355)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:490)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:408)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:166)
	at org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:118)
```

Proxy log:
```
[INFO ] 2023-01-11 15:13:04.143 [Connection-4-ThreadExecutor] ShardingSphere-SQL - Logic SQL: select ?::money
[INFO ] 2023-01-11 15:13:04.144 [Connection-4-ThreadExecutor] ShardingSphere-SQL - Actual SQL: ds_0 ::: select ?::money ::: [-92233720368547758.08]
[ERROR] 2023-01-11 15:13:05.561 [Connection-4-ThreadExecutor] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
org.postgresql.util.PSQLException: Bad value for type double : -92,233,720,368,547,758.08
	at org.postgresql.jdbc.PgResultSet.toDouble(PgResultSet.java:3314)
	at org.postgresql.jdbc.PgResultSet.getDouble(PgResultSet.java:2652)
	at org.postgresql.jdbc.PgResultSet.internalGetObject(PgResultSet.java:232)
	at org.postgresql.jdbc.PgResultSet.getObject(PgResultSet.java:2936)
	at com.zaxxer.hikari.pool.HikariProxyResultSet.getObject(HikariProxyResultSet.java)
	at org.apache.shardingsphere.infra.executor.sql.execute.result.query.impl.driver.jdbc.type.stream.JDBCStreamQueryResult.getValue(JDBCStreamQueryResult.java:106)
	at org.apache.shardingsphere.infra.merge.result.impl.transparent.TransparentMergedResult.getValue(TransparentMergedResult.java:43)
	at org.apache.shardingsphere.proxy.backend.communication.DatabaseCommunicationEngine.getRowData(DatabaseCommunicationEngine.java:348)
	at org.apache.shardingsphere.proxy.backend.handler.data.impl.UnicastDatabaseBackendHandler.getRowData(UnicastDatabaseBackendHandler.java:88)
	at org.apache.shardingsphere.proxy.frontend.postgresql.command.query.extended.Portal.nextPacket(Portal.java:169)
	at org.apache.shardingsphere.proxy.frontend.postgresql.command.query.extended.Portal.execute(Portal.java:145)
	at org.apache.shardingsphere.proxy.frontend.postgresql.command.query.extended.execute.PostgreSQLComExecuteExecutor.execute(PostgreSQLComExecuteExecutor.java:44)
	at org.apache.shardingsphere.proxy.frontend.postgresql.command.query.extended.execute.PostgreSQLComExecuteExecutor.execute(PostgreSQLComExecuteExecutor.java:34)
	at org.apache.shardingsphere.proxy.frontend.postgresql.command.query.extended.PostgreSQLAggregatedCommandExecutor.execute(PostgreSQLAggregatedCommandExecutor.java:41)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:110)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:77)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

### Reason analyze (If you can)

Proxy using `getObject` to retrieve value, while `money` type should be retrieve by `getString`.

https://github.com/apache/shardingsphere/blob/a4c01712aba7f17f3f4fe8264aaf4c19cb572432/proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/DatabaseCommunicationEngine.java#L348

PostgreSQL JDBC related issue: https://github.com/pgjdbc/pgjdbc/issues/425

### Example codes for reproduce this issue (such as a github link).

The following code worked if connected to PostgreSQL directly.
```java
try (PreparedStatement preparedStatement = connection.prepareStatement("select ?::money")) {
preparedStatement.setObject(1, "-92233720368547758.08");
try (ResultSet resultSet = preparedStatement.executeQuery()) {
    resultSet.next();
    System.out.println(resultSet.getString(1));
}
```

--- a/distribution/proxy/src/main/release-docs/LICENSE
+++ b/distribution/proxy/src/main/release-docs/LICENSE
@@ -333,7 +333,7 @@ The text of each license is also included at licenses/LICENSE-[project].txt.
     commons-compiler 3.1.11: https://github.com/janino-compiler/janino, BSD-3-Clause
     janino 3.1.11: https://github.com/janino-compiler/janino, BSD-3-Clause
     opengauss-jdbc 3.1.0-og: https://gitee.com/opengauss/openGauss-connector-jdbc, BSD-2-Clause
-    postgresql 42.7.2: https://github.com/pgjdbc/pgjdbc, BSD-2-Clause
+    postgresql 42.7.5: https://github.com/pgjdbc/pgjdbc, BSD-2-Clause
     protobuf-java 3.21.12: https://github.com/protocolbuffers/protobuf/blob/master/java, BSD-3-Clause
     protobuf-java-util 3.21.12: https://github.com/protocolbuffers/protobuf/blob/master/java, BSD-3-Clause
     stax2-api 4.2.1: https://github.com/FasterXML/stax2-api, BSD-2-Clause

--- a/docs/document/content/user-manual/shardingsphere-jdbc/optional-plugins/testcontainers/_index.cn.md
+++ b/docs/document/content/user-manual/shardingsphere-jdbc/optional-plugins/testcontainers/_index.cn.md
@@ -27,7 +27,7 @@ ShardingSphere 默认情况下不提供对 `org.testcontainers.jdbc.ContainerDat
     <dependency>
         <groupId>org.postgresql</groupId>
         <artifactId>postgresql</artifactId>
-        <version>42.7.2</version>
+        <version>42.7.5</version>
     </dependency>
     <dependency>
         <groupId>org.testcontainers</groupId>

--- a/docs/document/content/user-manual/shardingsphere-jdbc/optional-plugins/testcontainers/_index.en.md
+++ b/docs/document/content/user-manual/shardingsphere-jdbc/optional-plugins/testcontainers/_index.en.md
@@ -27,7 +27,7 @@ the possible Maven dependencies are as follows,
     <dependency>
         <groupId>org.postgresql</groupId>
         <artifactId>postgresql</artifactId>
-        <version>42.7.2</version>
+        <version>42.7.5</version>
     </dependency>
     <dependency>
         <groupId>org.testcontainers</groupId>

--- a/docs/document/content/user-manual/shardingsphere-proxy/optional-plugins/seata-at/_index.cn.md
+++ b/docs/document/content/user-manual/shardingsphere-proxy/optional-plugins/seata-at/_index.cn.md
@@ -332,7 +332,7 @@ CREATE SHARDING TABLE RULE t_order (
 <dependency>
     <groupId>org.postgresql</groupId>
     <artifactId>postgresql</artifactId>
-    <version>42.7.4</version>
+    <version>42.7.5</version>
 </dependency>
 ```
 

--- a/docs/document/content/user-manual/shardingsphere-proxy/optional-plugins/seata-at/_index.en.md
+++ b/docs/document/content/user-manual/shardingsphere-proxy/optional-plugins/seata-at/_index.en.md
@@ -337,7 +337,7 @@ The required JDBC Driver corresponds to the `proxy-frontend-database-protocol-ty
 <dependency>
     <groupId>org.postgresql</groupId>
     <artifactId>postgresql</artifactId>
-    <version>42.7.4</version>
+    <version>42.7.5</version>
 </dependency>
 ```
 

--- a/examples/pom.xml
+++ b/examples/pom.xml
@@ -42,7 +42,7 @@
         <spring-boot.version>2.3.12.RELEASE</spring-boot.version>
         <hikari-cp.version>3.4.2</hikari-cp.version>
         <mysql-connector-java.version>8.3.0</mysql-connector-java.version>
-        <postgresql.version>42.7.2</postgresql.version>
+        <postgresql.version>42.7.5</postgresql.version>
         <h2.version>2.2.224</h2.version>
         <slf4j.version>1.7.7</slf4j.version>
         <logback.version>1.2.13</logback.version>

--- a/pom.xml
+++ b/pom.xml
@@ -121,7 +121,7 @@
         <lombok.version>1.18.36</lombok.version>
         <immutables.version>2.9.3</immutables.version>
         
-        <postgresql.version>42.7.2</postgresql.version>
+        <postgresql.version>42.7.5</postgresql.version>
         <mysql-connector-java.version>8.3.0</mysql-connector-java.version>
         <mssql.version>6.1.7.jre8-preview</mssql.version>
         <h2.version>2.2.224</h2.version>

